src/SocketContext.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { useAuth } from './AuthProvider';
+import NotificationModal from '../src/components/common/notificationModal';
+
+const SocketContext = createContext();
+
+const setupSocketListeners = (socketIo, userId, setNotificationMessage, setIsModalOpen) => {
+  const handleNotification = (data) => {
+    console.log('Received notification:', data.content);
```

Comment:
do not forget remove unintended console logs before merge into main

Reply:
Thank you!
✅ Done